lib/elixir/lib/regex.ex

```diff
@@ -600,7 +600,7 @@ defmodule Regex do
       ["abc"]
 
       iex> Regex.split(~r//, "abc")
-      ["", "a", "b", "c", ""]
+      ["a", "b", "c", ""]
```

Comment:
As mentioned [here](https://github.com/elixir-lang/elixir/issues/14467#issuecomment-2849071034), this might be a breaking change, but is consistent with `:re.split/2`